client/my-sites/themes/index.js

```diff
@@ -6,37 +6,29 @@ import userFactory from 'lib/user';;
 import { navigation, siteSelection } from 'my-sites/controller';
 import { singleSite, multiSite, loggedOut, details, makeLoggedOutLayout } from './controller';
 
-const user = userFactory();
+export default function( router, renderer ) {
+	const user = userFactory();
+	const isLoggedIn = !! user.get();
 
-const isLoggedIn = !! user.get();
-
-const designRoutes = isLoggedIn
-	? {
-		'/design': [ multiSite, navigation, siteSelection ],
-		'/design/:site_id': [ singleSite, navigation, siteSelection ],
-		'/design/type/:tier': [ multiSite, navigation, siteSelection ],
-		'/design/type/:tier/:site_id': [ singleSite, navigation, siteSelection ],
+	if ( config.isEnabled( 'manage/themes' ) ) {
+		if ( isLoggedIn ) {
+			router( '/design', multiSite, navigation, siteSelection );
+			router( '/design/:site_id', singleSite, navigation, siteSelection );
+			router( '/design/type/:tier', multiSite, navigation, siteSelection );
+			router( '/design/type/:tier/:site_id', singleSite, navigation, siteSelection );
+		} else {
+			router( '/design', loggedOut, makeLoggedOutLayout );
+			router( '/design/type/:tier', loggedOut, makeLoggedOutLayout );
+		}
+		router( '/design*', renderer );
```

Comment:
Redirects, possibly?

What's more important is that this allows better separation of concerns with routing. Up to this PR, the renderer was magically attached as last middleware in each individual `router` invocation. That's an extra constraint imposed on individual sections' developers to keep in mind.

Now, we should be able to do things like

```js
if ( isLoggedIn ) {
	router( '/design', multiSite );
	router( '/design/:site_id', singleSite );
	router( '/design/type/:tier', multiSite );
	router( '/design/type/:tier/:site_id', singleSite );
	router( '/design*', navigation, siteSelection );
}
```

again, i.e. use catch-all middleware that is invoked for all subroutes -- and other sections' authors possibly relying on that `page` feature don't have to change their routing round.

(Maybe I should actually push a commit perusing that catch-all technique again?)